File: Minsk/CodeAnalysis/Syntax/BinaryExpressionSyntax.cs
namespace Minsk.CodeAnalysis.Syntax;

public class BinaryExpressionSyntax : ExpressionSyntax
{
    public BinaryExpressionSyntax(ExpressionSyntax left, SyntaxToken operatorToken, ExpressionSyntax right)
    {
        Left = left;
        OperatorToken = operatorToken;
        Right = right;
    }

    public override SyntaxKind Kind => SyntaxKind.BinaryExpression;

    public override IEnumerable<SyntaxNode> Children
    {
        get
        {
            yield return Left;
            yield return OperatorToken;
            yield return Right;
        }
    }

    public SyntaxToken OperatorToken { get; }
    public ExpressionSyntax Left { get; }
    public ExpressionSyntax Right { get; }
}

File: kt/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/BinaryExpressionSyntax.kt
package dev.phytolizer.minsk.analysis.syntax

class BinaryExpressionSyntax(val left: ExpressionSyntax, val operatorToken: SyntaxToken, val right: ExpressionSyntax) :
    ExpressionSyntax() {

    override val kind: SyntaxKind
        get() = SyntaxKind.BinaryExpression
    override val children: List<SyntaxNode>
        get() = listOf(left, operatorToken, right)
}
